pandas/tests/groupby/test_categorical.py

```diff
@@ -1193,3 +1193,23 @@ def test_groupby_categorical_axis_1(code):
     result = df.groupby(cat, axis=1).mean()
     expected = df.T.groupby(cat, axis=0).mean().T
     assert_frame_equal(result, expected)
+
+
+def test_groupby_cat_preserves_structure():
+    # GH 28787
+    df = DataFrame({"Name": ["Bob", "Greg"], "Item": [1, 2]})
+    expected = (
+        df.groupby("Name", observed=True)
+        .agg(pd.DataFrame.sum, skipna=True)
+        .reset_index()
+    )
+    expected["Name"] = expected["Name"].astype("category")
```

Comment:
I would like ``expected`` to be more specific, so

```python
df = DataFrame({"Name": Categorical(["Bob", "Greg"]), "Item":  [1, 2]})
expected = df.copy()
result = (
    df.groupby("Name", observed=True)
    .agg(pd.DataFrame.sum, skipna=True)
    .reset_index()
)
assert_frame_equal(result, expected)
```

Also, instead of using ``observed=True``, use the ``observed`´ fixture (it has already have been created). There isn't any skipna fixture, so that part can stay as it is.

Reply:
Makes sense, updated